lib/miq_automation_engine/engine/miq_ae_workspace.rb

```diff
@@ -63,23 +59,11 @@ def readonly?
       @readonly
     end
 
-    def self.instantiate_readonly(uri, workspace = nil, root = nil)
-      workspace ||= MiqAeWorkspaceRuntime.new(:readonly => true)
-      self._instantiate(uri, workspace, root)
-    end
-
-    def self.instantiate(uri, workspace = nil, root = nil)
-      workspace ||= MiqAeWorkspaceRuntime.new
```

Comment:
I was thinking of this [use case](https://github.com/ManageIQ/manageiq/blob/master/lib/miq_automation_engine/engine/miq_ae_service.rb#L133).  But it seems that a more elegant solution was introduced over time to call the instance method, instead of a class method passing in the workspace.  So, I am :+1: with your rationale.